Ride-js compilation tests

Kseolis opened the pull request.

Commit:
GetBinary.scala test

Commit:
updated jspi jsapiutils and test-js packages

Commit:
Merge branch 'NODE-2536' into ridejs-tests

Commit:
added first compilation tests, helpers class:
GeneratorContractsForBuiltInFunctions.scala
TestDataConstants.scala
RandomDataGenerator.scala

Commit:
added accountDataStorage tests
added randomInt for RandomDataGenerator.scala
actualVersions moved to TestDataConstants

StdLibVersion - added type annotation for key

Commit:
added new tests files

Commit:
added tests for AddressFromRecipient

Commit:
changed accesses to variables

Commit:
added tests for com/wavesplatform/test/builtInFunctions/blockchain

Commit:
Fixing test data retrieval

Commit:
added tests for com/wavesplatform/test/builtInFunctions/byteArray

Commit:
added tests for /converting
added NegativeTestsOfTheRideVersionsWithoutBigInt.scala

Commit:
added ToString.scala tests

Commit:
added TransferTransactionFromProto tests

Commit:
edited GetBinary names tests

Commit:
added tests for dappToDappInvocation, dataTransaction

Commit:
added private mod

Commit:
added decoding, encoding, hashing tests
rename NegativeTestsOfTheRideVersionsWithoutBigInt to
 NegativeTestsOfUnsupportedMethods

Commit:
added list methods tests

Commit:
edited ContainsElement

Commit:
added Fraction tests for test/builtInFunctions/math

Commit:
added Log tests for test/builtInFunctions/math
make testData protected

Commit:
added Median tests for test/builtInFunctions/math

Commit:
added Pow tests for test/builtInFunctions/math

Commit:
added Sqrt tests for test/builtInFunctions/math

Commit:
added tests for test/builtInFunctions/string/makeStringFunctions
and test/builtInFunctions/string/splitFunctions

Commit:
added Contains tests

Commit:
added Drop and DropRight tests

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/string/Size tests

Commit:
added
src/test/scala/com/wavesplatform/test/builtInFunctions/string/Take
 TakeRight tests

Commit:
added
src/test/scala/com/wavesplatform/test/builtInFunctions/string/IndexOf
 LastIndexOf tests

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/union/Extract

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/union/Value

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/union/ValueOrElse

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/union/ValueOrErrorMessage$

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/union/IsDefined

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/verification/rsaVerify tests

Commit:
added src/test/scala/com/wavesplatform/test/builtInFunctions/verification/sigVerify tests

Commit:
added Bn256Groth16Verify tests

Commit:
CreateMerkleRoot tests have been added to the src/test/scala/com/wavesplatform/test/builtInFunctions/verification directory

Commit:
Ecrecover tests have been added to the src/test/scala/com/wavesplatform/test/builtInFunctions/verification directory

Commit:
Groth16Verify tests have been added to the src/test/scala/com/wavesplatform/test/builtInFunctions/verification directory

Commit:
CheckMerkleProof tests have been added to the src/test/scala/com/wavesplatform/test/builtInFunctions/verification directory

yvaskov committed:
Added data storage tests for 'this'

yvaskov committed:
Added IsDataStorageUntouched tests for 'this'

yvaskov committed:
Added tests for blockchain functions

yvaskov committed:
Merge branch 'version-1.4.x' into ridejs-tests

yvaskov committed:
Merge branch 'ridejs-tests' into ridejs-additional-tests

yvaskov committed:
Optimized imports

yvaskov committed:
Optimized imports

yvaskov committed:
Merge remote-tracking branch 'origin/ridejs-tests' into ridejs-tests

xrtm000 commented on lang/tests-js/src/test/scala/com/wavesplatform/test/builtInFunctions/NegativeTestsOfUnsupportedMethods.scala:
```diff
@@ -0,0 +1,174 @@
+  private val min = "min(callerTestData)"
+
+  val tests: Tests = Tests {
+    test.apply("compilation error: Undefined type: `BigInt` for ride v3, v4") {
```
`test.apply(` could be simplified to `test(`

yvaskov replied:
Done

yvaskov committed:
Removed .apply from test

Commit:
tests refactoring for accountDataStorage

Commit:
refactoring tests from /test/builtInFunctions/blockchain

Commit:
refactoring tests from /test/builtInFunctions/byteArray

Commit:
small fix tests from /test/builtInFunctions/byteArray

Commit:
refactoring tests from /test/builtInFunctions/converting

Commit:
refactoring
ToUtf8String.scala
TransferTransactionFromProto.scala
tests from /test/builtInFunctions/converting

Commit:
refactoring tests from /dappToDappInvocation

Commit:
refactoring tests from /dataTransaction

Commit:
refactoring tests from decoding,
encoding, hashing, list

Commit:
refactoring tests from /builtInFunctions/math

Commit:
refactoring tests from /builtInFunctions/string
and /builtInFunctions/union

Commit:
refactoring tests from /builtInFunctions/verification
and NegativeTestsOfUnsupportedMethods

yvaskov committed:
Merge branch 'version-1.4.x' into ridejs-tests

# Conflicts:
#	lang/js/src/main/scala/com/wavesplatform/JsApiUtils.scala

yvaskov committed:
Merged 1.4.x, optimized imports, fixed AssetInfo test

yvaskov committed:
Merge branch 'version-1.4.x' into ridejs-tests

Commit:
fix codestyle for JsAPI.scala,
StdLibVersion.scala

Commit:
Merge remote-tracking branch 'origin/ridejs-tests' into ridejs-tests

phearnot committed:
Merge branch 'version-1.4.x' into ridejs-tests
